fastly/bigquery.go

```diff
@@ -3,36 +3,52 @@ package fastly
 import (
 	"fmt"
 	"net/url"
+	"sort"
+	"time"
 )
 
-// BigQuery represents a BigQuery logging response from the Fastly API.
+// BigQuery represents a BigQuery response from the Fastly API.
 type BigQuery struct {
-	ServiceID         string `mapstructure:"service_id"`
-	Name              string `mapstructure:"name"`
-	Format            string `mapstructure:"format"`
-	User              string `mapstructure:"user"`
-	ProjectID         string `mapstructure:"project_id"`
-	Dataset           string `mapstructure:"dataset"`
-	Table             string `mapstructure:"table"`
-	Template          string `mapstructure:"template_suffix"`
-	SecretKey         string `mapstructure:"secret_key"`
-	CreatedAt         string `mapstructure:"created_at"`
-	UpdatedAt         string `mapstructure:"updated_at"`
-	DeletedAt         string `mapstructure:"deleted_at"`
-	ResponseCondition string `mapstructure:"response_condition"`
-	Placement         string `mapstructure:"placement"`
+	ServiceID string `mapstructure:"service_id"`
+	Version   int    `mapstructure:"version"`
+
+	Name              string     `mapstructure:"name"`
+	Format            string     `mapstructure:"format"`
+	User              string     `mapstructure:"user"`
+	ProjectID         string     `mapstructure:"project_id"`
+	Dataset           string     `mapstructure:"dataset"`
+	Table             string     `mapstructure:"table"`
+	Template          string     `mapstructure:"template_suffix"`
+	SecretKey         string     `mapstructure:"secret_key"`
+	ResponseCondition string     `mapstructure:"response_condition"`
+	Placement         string     `mapstructure:"placement"`
+	FormatVersion     uint       `mapstructure:"format_version"`
+	CreatedAt         *time.Time `mapstructure:"created_at"`
```

Comment:
Do we want to change the types for the date fields now? This won't be backwards compatible.

Reply:
Why do we want to change them? If there is a need to change them then I am happy to open a new PR to change the metadata types for all other resources as well.

Reply:
It would say that nicer if we resolve the other ones as well, then we have just one version that breaks compatibility instead of multiple ones. I think that are only a few other cases. I will create an issue for these particularly.